### ISSUE TITLE
Generate accessor ignores the root backslash

### ISSUE THREAD

**Issue by Einenlum** [phpactor/phpactor#788]
Given:

```php
<?php

namespace App\Foo;

class Bar
{
    private $te<>st;

    public function __construct()
    {
        $this->test = new \DateTime();
    }
}
```

Then:

```php
<?php

namespace App\Foo;

class Bar
{
    private $test;

    public function __construct()
    {
        $this->test = new \DateTime();
    }

    public function test(): DateTime
    {
        return $this->test;
    }
}
```

It should be `\DateTime` as return type and not `DateTime`.

**Comment by elythyr** [comment]
I think it could be related to / treated with #742

**Comment by dantleech** [comment]
Sorry for not seeing this -- I would say the current behavior in this instance should be to import the `DateTime` class. Phpactor's current opinion is to import classes rather than use FQNs (even root classes).

So, I would prefer to do that by default -- it would be good to make this configurable: at _least_ in this specific case with "root" classes.

**Comment by Einenlum** [comment]
Okay, let's do this by default (import) and then later add a way to configure it :+1:

**Comment by Einenlum** [comment]
Actually while writing a test for this, it made me think.

```php
// File: source

<?php

namespace Acme;

class generateMethod
{
    /**
     * @var \DateTime
     */
    private $date;
<>}
// File: expected
<?php

namespace Acme;

use DateTime;

class generateMethod
{
    /**
     * @var DateTime
     */
    private $date;

    public function date(): DateTime
    {
        return $this->date;
    }
}
```
So that would be the expected behavior? To remove the trailing slash from the user's code and add an import statement?

**Comment by dantleech** [comment]
I guess the (un?)expected behavior would be to add the class import but leave the user's declaration as it was (there is currently no easy way of updating annotations anyway).

Actually, I think there was a worse reflection issue related to this -- around adding the ability to preserve the expansion state of a declared type -- with that you could determine if a class was fully or relatively "qualified", and then calculate its expansion in a given namespace.

So, f.e. `namespace Zed { ... function foobar(): Bar\Foo }`. The type `Foo` would be flagged as "expanded" or similar, then you would have both the types FQN `Zed\Bar\Foo` and the context namespace `Zed`, so client code could determine to render the type as `Bar\Foo` - relative to the context.

Is that the right kind of thing?

**Comment by Einenlum** [comment]
Yeah seems like a really interesting way to deal with it. But for now we can just add the use statement and let the previous code untouched.
I will try to make a PR when I have time. :)

**Comment by Einenlum** [comment]
So should I add the use statement only if was a FQCN type?

```php
// File: source

<?php

namespace Foo;

class Bar
{
    /** @var Baz */
    private $baz;

    /** @var \DateTime */
    private $date;

    /** @var \Some\Thing */
    private $thing;
}
// File: expected
<?php

namespace Foo;

use DateTime;
use Thing;

class Bar
{
    /** @var Baz */
    private $baz;

    /** @var \DateTime */
    private $date;

    /** @var \Some\Thing */
    private $thing;

    public function baz(): Baz
    {
        return $this->baz;
    }

    public function date(): DateTime
    {
        return $this->date;
    }
    public function thing(): Thing
    {
        return $this->thing;
    }
}
```

**Comment by dantleech** [comment]
hmm, yes I suppose so. The assumption is that all types should be imported, so I supose it follows that we import them... but then I'm wondering if it wouldn't be better to just support "qualfified" names.

Do you think fixing as above would add any value? or shall we concentrate on fixing the underlying (and more involved issue?
